Bottom: 29cf8618b6309dd170695d2d84112d29928931e7
Top:    7d318552e5adae4e28d33acb8af714df51e62087
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-10-27 15:53:48 +0800

Refresh of tuntap-convert-to-napi

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index ab9c104..5dc128a 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -454,12 +454,6 @@ static void handle_tx(struct vhost_net *net)
 			msg.msg_control = NULL;
 			ubufs = NULL;
 		}
-		if (!vhost_vq_avail_empty(&net->dev, vq) &&
-		    vq->delayed <= 16) {
-			vq->delayed++;
-			msg.msg_flags |= MSG_MORE;
-		} else
-			vq->delayed = 0;
 		/* TODO: Check specific error and bomb out unless ENOBUFS? */
 		err = sock->ops->sendmsg(sock, &msg, len);
 		if (unlikely(err < 0)) {
diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 8214e33..c6f2d89 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -311,7 +311,6 @@ static void vhost_vq_reset(struct vhost_dev *dev,
 	vq->busyloop_timeout = 0;
 	vq->umem = NULL;
 	vq->iotlb = NULL;
-	vq->delayed = 0;
 }
 
 static int vhost_worker(void *data)
diff --git a/drivers/vhost/vhost.h b/drivers/vhost/vhost.h
index bff28a0..78f3c5f 100644
--- a/drivers/vhost/vhost.h
+++ b/drivers/vhost/vhost.h
@@ -132,7 +132,6 @@ struct vhost_virtqueue {
 	/* Log write descriptors */
 	void __user *log_base;
 	struct vhost_log *log;
-	u16 delayed;
 
 	/* Ring endianness. Defaults to legacy native endianness.
 	 * Set to true when starting a modern virtio device. */
